src/boot/efi/util.c

```diff
@@ -403,17 +408,46 @@ void sort_pointer_array(
 
         for (size_t i = 1; i < n_members; i++) {
                 size_t k;
-                void *entry = array[i];
+                uint8_t tmp[stride];
```

Comment:
hmm, VLAs suck. we generally try to avoid them these days. just use xmalloc() once?